packages/shared/trade/src/transaction-sidebar/transaction-sidebar.tsx

```diff
@@ -104,10 +135,26 @@ export const TransactionSidebar = ({
     }
   };
 
-  return confirm ? (
-    <Confirmation2
+  const handleActionSidebarCancel = () => {
+    if (onCancelCallback) {
+      onCancelCallback();
+    }
+    if (onCancelRouteCallback) {
+      onCancelRouteCallback();
+    }
+  };
+
+  return showTxnApprovals ? (
+    <TransactionApprovals
+      context={context}
+      onCancel={() => setShowTxnApprovals(false)}
+      showDrawer={isPortfolio ? false : showDrawer === true}
```

Comment:
Can we just make `showDrawer` a variable in the outer scope instead of doing it three times here